packages/compass-user-data/src/user-data.ts

```diff
@@ -68,6 +69,7 @@ export class UserData<T extends z.Schema> {
   private readonly serialize: SerializeContent<z.input<T>>;
   private readonly deserialize: DeserializeContent;
   private readonly getFileName: GetFileName;
+  private readonly semaphore = new Semaphore(1000);
```

Comment:
The Node.js thread pool for fs operations (`UV_THREADPOOL_SIZE`) has a default size of 4 or so. I think a lower number, say 50 or 100, would also be fine here and result in similar performance while at the same time still being significantly better at reducing the risk of EMFILE/ENFILE errors